File: src/common/com.smi.microservices.common/src/main/java/org/smi/common/options/ConsumerOptions.java

package org.smi.common.options;

/**
 * Class for managing the options for a RabbitMQ consumer
 */
public class ConsumerOptions {

	/**
	 * Name of the queue to consume from
	 */
	public String QueueName;

	/**
	 * Max number of messages the queue will send the consumer before \ receiving an
	 * acknowledgement
	 */
	public int QoSPrefetchCount = 1;

	/**
	 * Automatically acknowledge any messages sent to the consumer
	 */
	public boolean AutoAck = false;

	//TODO VerifyPopulated
}

File: src/SmiServices/Common/Options/ConsumerOptions.cs
using System.Text;

namespace SmiServices.Common.Options;

/// <summary>
/// Configuration options needed to receive messages from a RabbitMQ queue.
/// </summary>
public class ConsumerOptions : IOptions
{
    /// <summary>
    /// Name of the queue to consume from.
    /// </summary>
    public string? QueueName { get; set; }

    /// <summary>
    /// Max number of messages the queue will send the consumer before receiving an acknowledgement.
    /// </summary>
    public ushort QoSPrefetchCount { get; set; } = 1;

    /// <summary>
    /// Automatically acknowledge any messages sent to the consumer.
    /// </summary>
    public bool AutoAck { get; set; }

    /// <summary>
    /// If set, consumer will not call Fatal when an unhandled exception occurs when processing a message, up to the <see cref="QoSPrefetchCount"/> limit. Requires <see cref="AutoAck"/> to be false
    /// </summary>
    public bool HoldUnprocessableMessages { get; set; }


    /// <summary>
    /// Verifies that the individual options have been populated
    /// </summary>
    /// <returns></returns>
    public bool VerifyPopulated()
    {
        return !string.IsNullOrWhiteSpace(QueueName) && (QoSPrefetchCount != 0);
    }

    public override string ToString()
    {
        var sb = new StringBuilder();
        sb.Append("QueueName: " + QueueName);
        sb.Append(", AutoAck: " + AutoAck);
        sb.Append(", QoSPrefetchCount: " + QoSPrefetchCount);
        sb.Append(", HoldUnprocessableMessages: " + HoldUnprocessableMessages);
        return sb.ToString();
    }
}
